bluemira/base/designer.py

```diff
@@ -55,11 +58,20 @@ def __init__(
         self,
         params: Union[Dict, ParameterFrame, ConfigParams, None],
         build_config: Optional[Dict] = None,
+        *,
+        verbose=True,
     ):
         self.params = make_parameter_frame(params, self.param_cls)
         self.build_config = build_config if build_config is not None else {}
         self._run_func = self._get_run_func(self.run_mode)
 
+        self.execute = timing(
```

Comment:
I think we're going to lose the typing on `execute` doing this (and in `Builder.build`). As `timing` returns a vanilla `Callable`, type checkers will now think `execute` just returns `Any`.

The only thing I can think of is to use a `TypeVar` in the `timing` signature. Something like this.

```python
_T = TypeVar("_T")
def timing(func: Callable[..., _T], ...) -> Callable[..., _T]:
```

This at least gets us the return type. I can't work out how to get it to recognise the argument types, though.

There may be alternative solutions, though. Would a decorator work? Or did you try that?

Reply:
It could be used as a decorator here but we will lose all the customisation.  I've added `wraps` which should help and changed the location of the wrapping for the designer. That should take care of the typing for the designer. For the builder I dont really know the best way as the user writes the `build` function